resources/fetching/registry/registry_test.go

```diff
@@ -256,3 +260,97 @@ func (c *boolFetcherCondition) Condition() bool {
 func (c *boolFetcherCondition) Name() string {
 	return c.name
 }
+
+func Test_registry_Update(t *testing.T) {
```

Comment:
Can you add another test where you initial the registry with no `Option`s?
The empty use case should probably just returns an error or hold an empty map that cannot be changed.
I want to make sure it doesn't panic.

Reply:
That would be pretty much equivalent to the `check nil` case